app/commands/v2/patch_link_set.rb

```diff
@@ -25,19 +25,38 @@ def call
         # we need to reload the link_set as the links association will be stale
         link_set.reload
 
-        orphaned_content_ids = link_diff_between(links_before_patch, link_set.links.map(&:target_content_id))
+        orphaned_content_ids = link_diff_between(before_links.map(&:target_content_id), link_set.links.map(&:target_content_id))
 
         after_transaction_commit do
           send_downstream(orphaned_content_ids)
         end
 
         Action.create_patch_link_set_action(link_set, event)
 
+        record_link_changes(before_links, link_set.links.to_a)
+
         presented = Presenters::Queries::LinkSetPresenter.present(link_set)
         Success.new(presented)
       end
 
-    private
+      private
+
+      def record_link_changes(before_links, after_links)
```

Comment:
I think this method is sufficiently complex to pull out into a class. That may make it easier to refactor too.